Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator.swift

```diff
@@ -148,11 +148,28 @@ public actor SwiftSDKGenerator {
     )
   }
 
+  func doesPathExist(
+    _ containerPath: FilePath,
+    inContainer id: String
+  ) async throws -> Bool {
+    let result = try await Shell.readStdout(
+      #"\#(Self.dockerCommand) exec \#(id) sh -c 'test -e "\#(containerPath)" && echo "y" || echo "n"'"#,
+      shouldLogCommands: self.isVerbose
+    )
+    .trimmingCharacters(in: .whitespacesAndNewlines)
+    return result == "y"
```

Comment:
I think you can avoid the `echo` and string comparison with something like this:
```suggestion
    do {
      try await Shell.run(
        #"\#(Self.dockerCommand) exec \#(id) sh -c 'test -e "\#(containerPath)"'"#,
        shouldLogCommands: self.isVerbose
      )
      return true
    } catch FileOperationError.nonZeroExitCode {
      return false
    }
```

Reply:
@euanh but this would mask other errors (like if the container does not exist or does not run), right? I thought about this too, but wanted to make sure I'm not masking other errors.
Although, they might arguably show up again shortly after, or even before this.

Reply:
> @euanh but this would mask other errors (like if the container does not exist or does not run), right? I thought about this too, but wanted to make sure I'm not masking other errors. Although, they might arguably show up again shortly after, or even before this.

Yes, that's true.  `doesPathExist` / `copyFromDockerContainer` is currently only ever called within a `withDockerContainer` wrapper so it should not be reached if the container failed to start, but the `docker exec` might fail if the container has crashed in the meantime.   Also, nothing in the API currently enforces this restriction - it's possible to call `copyFromDockerContainer` from outside a wrapper, in which case the provided container ID might be completely invalid.